### ISSUE TITLE
Infinite unresponsive loop on debugging any extension

### ISSUE THREAD

**Issue by aciq** [microsoft/vscode#168817]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.74.0
- OS Version: Linux mint 21

Steps to Reproduce:

1. Press F5 to start debugging any vscode extension template (e.g from `yo code`) 
2. The following message appears in debug console
```
Could not read source map for file:///usr/share/code/resources/app/out/vs/out-vscode/vs/loader.js: ENOENT: no such file or directory, open '/usr/share/code/resources/app/out/vs/out-vscode/vs/nls.js.map'
```
3. This folder out-vscode does not exist
4. Vscode goes into an infinite cpu cycle and the only way to stop debugging is to kill the process

i'm using this (default) launch.json file to run the extension

```
{
	"version": "0.2.0",
	"configurations": [
		{
			"name": "Run Extension",
			"type": "extensionHost",
			"request": "launch",
			"args": [
				"--extensionDevelopmentPath=${workspaceFolder}"
			],
		}
	]
}
```

**Comment by aciq** [comment]
i think it might be linked to the current working directory when you start code, this infinite loop does not happen when code is started from the startup menu. 

however it does happen when you launch code from the terminal or a script

**Comment by roblourens** [comment]
Are there any environment variables set in that environment that are related to node/electron?

**Comment by aciq** [comment]
the names/count of environment vars is identical both when started from menu and when started from terminal. no variables related to node

**Comment by roblourens** [comment]
How did you determine that, from the terminal? That might have been sanitized or your shell might have overwritten the parent environment. Can you check the environment of vscode's extensionHost process from the outside (I think this is `/proc/<pid>/environ` or something similar)

Can you also just share the output of `printenv` in the terminal that you launch from that doesn't work?

And third, can you give the full command line of the process that is not responding, so I know which vscode process it is?

**Comment by aciq** [comment]
ohh you're right about the environment variables - i was trying that through the shell. ill get back to you in a moment

**Comment by aciq** [comment]
ok so in this example the process 34467 is out of control and here are the environment variables for this process.

![processes](https://user-images.githubusercontent.com/36763595/207190504-f2794118-49c7-454e-bd39-f1ec271d364e.png)

```bash
ian@ian-mint:/mnt/e/projects/aasdasd$ cat /proc/34467/environ
ELECTRON_RUN_AS_NODE=1LANGUAGE=en_USUSER=ianroc=/mnt/e/box1/proj/roc-lang/_roc/rocXDG_SEAT=seat0DOTNET_CLI_TELEMETRY_OPTOUT=1XDG_SESSION_TYPE=x11NEMO_SCRIPT_WINDOW_GEOMETRY=1280x1372+2560+0SHLVL=0HOME=/home/ianDESKTOP_SESSION=cinnamonGIO_LAUNCHED_DESKTOP_FILE=/usr/share/applications/nemo.desktopGTK_MODULES=gail:atk-bridgeXDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0LC_MONETARY=et_EE.UTF-8DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/busNEMO_SCRIPT_CURRENT_URI=file:///mnt/e/projects/fable-vscode-demoGIO_LAUNCHED_DESKTOP_FILE_PID=34329NEMO_SCRIPT_NEXT_PANE_SELECTED_FILE_PATHS=INSIDE_NEMO_PYTHON=MANDATORY_PATH=/usr/share/gconf/cinnamon.mandatory.pathQT_QPA_PLATFORMTHEME=qt5ctLOGNAME=ianNEMO_SCRIPT_SELECTED_URIS=XDG_SESSION_CLASS=userDEFAULTS_PATH=/usr/share/gconf/cinnamon.default.pathXDG_SESSION_ID=c2GTK_OVERLAY_SCROLLING=1DOTNET_ROOT=/usr/lib/dotnet/dotnet6-6.0.110GNOME_DESKTOP_SESSION_ID=this-is-deprecatedPATH=/home/ian/.dotnet/tools:/home/ian/bin/:/home/ian/.pack/bin:/mnt/e/box1/proj/fstar-lang/fstar/bin:/home/ian/.local/bin:/home/ian/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/ian/.dotnet/tools:/home/ian/.local/share/JetBrains/Toolbox/scriptsGDM_LANG=en_USGTK3_MODULES=xapp-gtk3-moduleSESSION_MANAGER=local/ian-mint:@/tmp/.ICE-unix/1845,unix/ian-mint:/tmp/.ICE-unix/1845XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session0LC_ADDRESS=et_EE.UTF-8XDG_RUNTIME_DIR=/run/user/1000DISPLAY=:0LANG=en_US.UTF-8DOTNET_BUNDLE_EXTRACT_BASE_DIR=/home/ian/.cache/dotnet_bundle_extractLC_TELEPHONE=et_EE.UTF-8XDG_CURRENT_DESKTOP=X-CinnamonXDG_SESSION_DESKTOP=cinnamonXAUTHORITY=/home/ian/.XauthorityXDG_GREETER_DATA_DIR=/var/lib/lightdm-data/ianSSH_AUTH_SOCK=/run/user/1000/keyring/sshSHELL=/bin/bashLC_NAME=et_EE.UTF-8DOTNET_NEW_PREFERRED_LANG=F#QT_ACCESSIBILITY=1GDMSESSION=cinnamonLC_MEASUREMENT=et_EE.UTF-8NEMO_SCRIPT_SELECTED_FILE_PATHS=GPG_AGENT_INFO=/run/user/1000/gnupg/S.gpg-agent:0:1LC_IDENTIFICATION=et_EE.UTF-8XDG_VTNR=7NEMO_SCRIPT_NEXT_PANE_CURRENT_URI=PWD=/mnt/e/projects/fable-vscode-demoXDG_CONFIG_DIRS=/etc/xdg/xdg-cinnamon:/etc/xdgXDG_DATA_DIRS=/usr/share/cinnamon:/usr/share/gnome:/home/ian/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/shareLC_NUMERIC=et_EE.UTF-8LC_PAPER=et_EE.UTF-8NEMO_SCRIPT_NEXT_PANE_SELECTED_URIS=VSCODE_CWD=/mnt/e/projects/fable-vscode-demoVSCODE_CLI=1ELECTRON_NO_ATTACH_CONSOLE=1CHROME_DESKTOP=code-url-handler.desktopORIGINAL_XDG_CURRENT_DESKTOP=X-CinnamonGDK_BACKEND=x11VSCODE_NLS_CONFIG={"locale":"en-us","availableLanguages":{},"_languagePackSupport":true}VSCODE_CODE_CACHE_PATH=/home/ian/.config/Code/CachedData/5235c6bb189b60b01b1f49062f4ffa42384f8c91VSCODE_IPC_HOOK=/run/user/1000/vscode-5f2b04ed-1.74.0-main.sockVSCODE_PID=34348VSCODE_AMD_ENTRYPOINT=vs/workbench/api/node/extensionHostProcessVSCODE_HANDLES_UNCAUGHT_ERRORS=trueVSCODE_EXTHOST_IPC_HOOK=/run/user/1000/vscode-ipc-16ad17c4-4880-49f1-83fa-ee0ebfcbc152.sockNODE_CHANNEL_FD=3NODE_CHANNEL_SERIALIZATION_MODE=json

```

**Comment by aciq** [comment]
and as a comparision, when i launch from the start menu it runs fine, has processes like this

![image](https://user-images.githubusercontent.com/36763595/207195508-25f93f82-7e08-40c8-8280-5a535fe5a99a.png)


and environ has these variables (i guessed the proc id by memory usage and variables)

```bash
ian@ian-mint:~$ cat /proc/37132/environ
ELECTRON_RUN_AS_NODE=1GJS_DEBUG_TOPICS=JS ERROR;JS LOGLANGUAGE=en_USUSER=ianroc=/mnt/e/box1/proj/roc-lang/_roc/rocXDG_SEAT=seat0DOTNET_CLI_TELEMETRY_OPTOUT=1XDG_SESSION_TYPE=x11SHLVL=0HOME=/home/ianDESKTOP_SESSION=cinnamonGIO_LAUNCHED_DESKTOP_FILE=/usr/share/applications/code.desktopGTK_MODULES=gail:atk-bridgeXDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0LC_MONETARY=et_EE.UTF-8DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/busCINNAMON_VERSION=5.4.12GIO_LAUNCHED_DESKTOP_FILE_PID=36145MANDATORY_PATH=/usr/share/gconf/cinnamon.mandatory.pathQT_QPA_PLATFORMTHEME=qt5ctLOGNAME=ianXDG_SESSION_CLASS=userDEFAULTS_PATH=/usr/share/gconf/cinnamon.default.pathXDG_SESSION_ID=c2GTK_OVERLAY_SCROLLING=1DOTNET_ROOT=/usr/lib/dotnet/dotnet6-6.0.110GNOME_DESKTOP_SESSION_ID=this-is-deprecatedPATH=/home/ian/.dotnet/tools:/home/ian/bin/:/home/ian/.pack/bin:/mnt/e/box1/proj/fstar-lang/fstar/bin:/home/ian/.local/bin:/home/ian/bin:/home/ian/.cargo/bin:/home/ian/.dotnet/tools:/home/ian/bin/:/home/ian/.pack/bin:/mnt/e/box1/proj/fstar-lang/fstar/bin:/home/ian/.local/bin:/home/ian/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/ian/.dotnet/tools:/home/ian/.local/share/JetBrains/Toolbox/scripts:/home/ian/.dotnet/tools:/home/ian/.local/share/JetBrains/Toolbox/scriptsGDM_LANG=en_USGTK3_MODULES=xapp-gtk3-moduleSESSION_MANAGER=local/ian-mint:@/tmp/.ICE-unix/1845,unix/ian-mint:/tmp/.ICE-unix/1845XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session0LC_ADDRESS=et_EE.UTF-8XDG_RUNTIME_DIR=/run/user/1000DISPLAY=:0LANG=en_US.UTF-8DOTNET_BUNDLE_EXTRACT_BASE_DIR=/home/ian/.cache/dotnet_bundle_extractLC_TELEPHONE=et_EE.UTF-8XDG_CURRENT_DESKTOP=X-CinnamonXDG_SESSION_DESKTOP=cinnamonXAUTHORITY=/home/ian/.XauthorityXDG_GREETER_DATA_DIR=/var/lib/lightdm-data/ianSSH_AUTH_SOCK=/run/user/1000/keyring/sshSHELL=/bin/bashLC_NAME=et_EE.UTF-8DOTNET_NEW_PREFERRED_LANG=F#QT_ACCESSIBILITY=1GDMSESSION=cinnamonLC_MEASUREMENT=et_EE.UTF-8GPG_AGENT_INFO=/run/user/1000/gnupg/S.gpg-agent:0:1LC_IDENTIFICATION=et_EE.UTF-8GJS_DEBUG_OUTPUT=stderrXDG_VTNR=7PWD=/home/ianXDG_CONFIG_DIRS=/etc/xdg/xdg-cinnamon:/etc/xdgXDG_DATA_DIRS=/usr/share/cinnamon:/usr/share/gnome:/home/ian/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/shareLC_NUMERIC=et_EE.UTF-8LC_PAPER=et_EE.UTF-8CHROME_DESKTOP=code-url-handler.desktopORIGINAL_XDG_CURRENT_DESKTOP=X-CinnamonVSCODE_CWD=/home/ianGDK_BACKEND=x11VSCODE_NLS_CONFIG={"locale":"en-us","availableLanguages":{},"_languagePackSupport":true}VSCODE_IPC_HOOK=/run/user/1000/vscode-5f2b04ed-1.74.0-main.sockVSCODE_PID=36145LS_COLORS=LESSCLOSE=/usr/bin/lesspipe %s %sLESSOPEN=| /usr/bin/lesspipe %s_=/usr/share/code/codeVSCODE_AMD_ENTRYPOINT=vs/workbench/api/node/extensionHostProcessVSCODE_HANDLES_UNCAUGHT_ERRORS=trueVSCODE_EXTHOST_IPC_HOOK=/run/user/1000/vscode-ipc-69a7b720-7f69-4c3e-94ad-c2ebf452c01a.sockNODE_CHANNEL_FD=3NODE_CHANNEL_SERIALIZATION_MODE=json
```

**Comment by roblourens** [comment]
Isn't that just showing 6% CPU though? And can you help me figure out which process it is either by giving the full command line (eg from `ps`) or finding it in the vscode process explorer (Help > Open Process Explorer)

**Comment by aciq** [comment]
found it, 
process 38218 this time

![ps](https://user-images.githubusercontent.com/36763595/207197979-99fa0b61-9d19-4602-b669-b31b83be7037.png)

![procs](https://user-images.githubusercontent.com/36763595/207197989-03c9b22c-7a23-4c04-b632-0f1f94db5169.png)

```bash
ian@ian-mint:~$ ps 38218
    PID TTY      STAT   TIME COMMAND
  38218 ?        Rl     1:18 /usr/share/code/code --ms-enable-electron-run-as-no
```

6% implies it is a single threaded javascript process running at 100% capacity, it couldn't go any higher than this

**Comment by roblourens** [comment]
Any guesses @connor4312?

**Comment by connor4312** [comment]
This was https://github.com/microsoft/vscode-js-debug/issues/1469 which should be fixed when you update to 1.74.1
